### ISSUE TITLE
Putting vanilla web components using Shadow DOM in a Stencil app fails on Firefox

### ISSUE THREAD

**Issue by LostInBrittany** [ionic-team/stencil#1047]
**Stencil version:**

```
 @stencil/core@0.11.0 
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
I am trying to use some non-stencil based web components (vanilla or based on lit-element or polymer 3). They works without problem on Chrome, but on Firefox the app fails with a `this.attachShadow is not a function` error. I have tested several versions of Firefox, including current stable (61.0.1) and nightly build (63.0 alpha 1).

![image](https://user-images.githubusercontent.com/726476/44792018-475b8800-aba3-11e8-8269-aff2a98216ec.png)

I thought Stencil is not currently polyfilling ShadowDOM on Firefox, so I added the `@webcomponents/webcomponentsjs` polyfill myself and loaded it from the `index.html`
either with `webcomponents-loader` or directly with the `webcomponents-sd.js` bundle, but then it fails other errors:

**Expected behavior:**

It should work as in Chrome.

![image](https://user-images.githubusercontent.com/726476/44794649-2e55d580-aba9-11e8-8753-a5cc6ac49632.png)

**Steps to reproduce:**

See next comment (I've split it to make it simple to read)

**Related code:**

See next comment (I've split it to make it easy to read)

**Other information:**

**Comment by LostInBrittany** [comment]
## Steps to reproduce


### Creating the test elements

I have created a very simple vanilla web component without Shadow DOM:

```js
class MyVanillaElement extends HTMLElement {
    connectedCallback() {
        let tmpl = document.createElement('template');
        tmpl.innerHTML = `<h1>I'm a vanilla custom element without Shadow DOM</h1>`;
        this.appendChild(tmpl.content.cloneNode(true));
    }
}
customElements.define('my-vanilla-element', MyVanillaElement);
```

and another one with Shadow DOM:

```js
class MyVanillaElementWithSadowDom extends HTMLElement {
    connectedCallback() {
        let tmpl = document.createElement('template');
        tmpl.innerHTML = `<h1>I'm a vanilla custom element with Shadow DOM</h1>`;
        const shadowRoot = this.attachShadow({ mode: 'open' });
        shadowRoot.appendChild(tmpl.content.cloneNode(true));
        console.log('shadowRoot', shadowRoot);
    }
}
customElements.define('my-vanilla-element-with-shadowdom', MyVanillaElementWithSadowDom);
```


### Using the elements in a Stencil app

I'm importing and using them in a very simple stencil element:

```tsx
import { Component } from '@stencil/core';

import '../../../non-stencil-components/my-vanilla-element';
import '../../../non-stencil-components/my-vanilla-element-with-shadowdom';

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true
})
export class MyComponent {
  render() {
    return (
      <div>
        <div>Let's illustrate the bug with a quick example:</div>
        <div><my-vanilla-element></my-vanilla-element></div>   
        <div><my-vanilla-element-with-shadowdom></my-vanilla-element-with-shadowdom></div>
      </div>
    );
  }
}
  ```

On Chrome it works as intended:

![image](https://user-images.githubusercontent.com/726476/44794649-2e55d580-aba9-11e8-8753-a5cc6ac49632.png)

On Firefox it fails:

![image](https://user-images.githubusercontent.com/726476/44794848-a6240000-aba9-11e8-8de2-c68132337787.png)

### Trying to use the Polyfill

Then I added the `@webcomponents/webcomponentsjs`  polyfill 

``` 
npm i @webcomponents/webcomponentsjs
```

I modified `stencil.config.ts` to copy it and its dependencies as resource (polyfill can't be used as a module right now :( ):

```
import { Config } from '@stencil/core';

export const config: Config = {
  namespace: 'mycomponent',
  copy: [
    { 
      src: "../node_modules/@webcomponents/webcomponentsjs/*.js", 
      dest: "node_modules/@webcomponents/webcomponentsjs"
    },
    { 
      src: "../node_modules/@webcomponents/webcomponentsjs/bundles/*", 
      dest: "node_modules/@webcomponents/webcomponentsjs/bundles"
    },
  ],
  outputTargets:[
    {
      type: 'dist'
    },
    {
      type: 'www',
      serviceWorker: null
    }
  ]
};
```

and then I loaded it in `index.html`:

```html
<!DOCTYPE html>
<html dir="ltr" lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
  <title>Stencil Component Starter</title>
  <script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
  <script src="/build/mycomponent.js"></script>

</head>
<body>

  <my-component first="Stencil" last="'Don't call me a framework' JS"></my-component>

</body>
</html>

```

It works as usual in Chrome but we get a blocking error in Firefox (`TypeError: "Argument 1 of MutationObserver.observe does not implement interface Node."`):

![image](https://user-images.githubusercontent.com/726476/44795366-c902e400-abaa-11e8-953a-2bb20759465b.png)

I guessed that the polyfill interferes with Stencil internal polyfilling, so I tried to flip the order (loading polyfill after stencil app, 


```html
...
   <script src="/build/mycomponent.js"></script>
  <script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
...

```

Still no problem in Chrome. In Firefox, it's complicated. It kinda works, but I get a `TypeError: "CustomElementRegistry" is read-only` error:

![image](https://user-images.githubusercontent.com/726476/44796004-5266e600-abac-11e8-8197-93b7ecf6447b.png)

**Comment by LostInBrittany** [comment]
## Related code

You can get the related code, based on Stencil component starter, on https://github.com/LostInBrittany/stencil-and-other-webcomponents-bug

**Comment by LostInBrittany** [comment]
ping ? :) :)

**Comment by manucorporat** [comment]
This should be fixed in stencil 1.0 which was merged recently in master!
